internal/kuberesource/sets.go

```diff
@@ -496,3 +497,128 @@ func Emojivoto(smMode serviceMeshMode) []any {
 
 	return resources
 }
+
+// VolumeStatefulSet returns a stateful set for testing volume mounts and the
+// mounting of encrypted luks volumes using the workload-secret.
+func VolumeStatefulSet() []any {
+	initCommand := `#!/bin/bash
+# cryptsetup complains if there is no /run directory.
+mkdir /run
+set -e
+# device is the path to the block device to be encrypted.
+device="/dev/csi0"
+# workload_secret_path is the path to the Contrast workload secret.
+workload_secret_path="/contrast/secrets/workload-secret-seed"
+# tmp_key_path is the path to a temporary key file.
+tmp_key_path="/dev/shm/key"
+# disk_encryption_key_path is the path to the disk encryption key.
+disk_encryption_key_path="/dev/shm/disk-key"
+
+# If the device is already a LUKS device, format it.
+if ! cryptsetup isLuks "${device}"; then
+	# Generate a random key for the first initialization.
```

Comment:
Have you considered removing the key slot with this key after initialization?

Reply:
Ok that was easier than imagined. I thought I needed multiple commands to find out and then delete the key slot. Apparently not.

Reply:
Excuse my ignorance: what's the benefit of doing this, as opposed to initializing with the derived key right away?

Reply:
We derive the key from the UUID of the LUKS partition and we don't know the UUID before we create it.